src/src/pages/capabilities/MyCapabilities.js

```diff
@@ -1,29 +1,50 @@
-import React, { useContext } from "react";
+import React, { useContext, useEffect, useState } from "react";
 import { Text } from "@dfds-ui/typography";
 import { useNavigate } from "react-router-dom";
 import { ChevronRight, StatusAlert } from "@dfds-ui/icons/system";
 import {
+  Spinner,
   Table,
-  TableHead,
   TableBody,
-  TableRow,
-  TableHeaderCell,
   TableDataCell,
+  TableHead,
+  TableHeaderCell,
+  TableRow,
 } from "@dfds-ui/react-components";
-import { Spinner } from "@dfds-ui/react-components";
 import AppContext from "AppContext";
 import PageSection from "components/PageSection";
-import styles from "./myCapabilities.css";
+import CapabilityCostSummary from "components/BasicCapabilityCost";
 
 export default function MyCapabilities() {
-  const { myCapabilities, appStatus } = useContext(AppContext);
+  const { myCapabilities, capabilityCosts, appStatus } = useContext(AppContext);
 
   const items = myCapabilities || [];
   const isLoading = !appStatus.hasLoadedMyCapabilities;
+  const isLoadingCosts = !appStatus.hasLoadedCosts;
+  const [showCostsSpinner, setShowCostsSpinner] = useState(isLoadingCosts);
 
   const navigate = useNavigate();
   const clickHandler = (id) => navigate(`/capabilities/${id}`);
 
+  useEffect(() => {
+    let isMounted = true;
+
+    const interval = setInterval(() => {
+      if (isMounted) {
+        setShowCostsSpinner(false);
+      }
+    }, 3000);
```

Comment:
This is not the most expensive method, but should we stop checking once isMounted has been true once?

Reply:
Having read through this function scope a few times, this seems odd. Perhaps it may just be that I've looked at it so many times that I've blinded myself 😅.

As far as I can see, the conditions of this never changes while it is in use. isMounted stays true until the MyCapabilities component gets cleaned up, as which point isMounted is explicitly set to false, but since the component is getting cleaned up, it doesn't matter.

Since I haven't tested this locally, perhaps I'm missing something, but it would seem the useEffect right below this one from line 45 till 47-ish, should take care of it.

Reply:
This issue was to prevent the issue where the spinner would be set after being unmounted